src/subhalo_tracking.cpp

```diff
@@ -104,7 +104,19 @@ struct CompareMass_t
   }
   bool operator()(const HBTInt &i, const HBTInt &j)
   {
-    return (*Subhalos)[i].Mbound > (*Subhalos)[j].Mbound;
+    const Subhalo_t &sub1 = (*Subhalos)[i];
+    const Subhalo_t &sub2 = (*Subhalos)[j];
+
+    // Try to compare by mass first
+    if(sub1.Mbound != sub2.Mbound)
+      return sub1.Mbound > sub2.Mbound;
+
+    // Where masses are equal, check vmax
+    if(sub1.VmaxPhysical != sub2.VmaxPhysical)
+      return sub1.VmaxPhysical > sub2.VmaxPhysical;
+    
+    // Otherwise fall back to using the TrackId, which is always unique
+    return sub1.TrackId > sub2.TrackId;
```

Comment:
This might not result in completely reproducible code, since the `TrackId` for the same object can change from across runs when using MPI.

Reply:
I'll make it use mass, vmax, mostboundid then trackid. The trackid should never be used if duplicates are eliminated but we can still test this branch without merging those changes.